hnswlib/hnswalg.h

```diff
@@ -799,11 +842,47 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
     }
 
 
-    /**
-        * Adds point. Updates the point if it is already in the index
+    /*
+    * Adds point. Updates the point if it is already in the index.
+    * If replacement of deleted elements is enabled: replaces previously deleted point if any, updating it with new point
     */
-    void addPoint(const void *data_point, labeltype label) {
-        addPoint(data_point, label, -1);
+    void addPoint(const void *data_point, labeltype label, bool replace_deleted = false) {
+        if ((replace_deleted_ == false) && (replace_deleted == true)) {
```

Comment:
Maybe rename `replace_deleted_` to `allow_replace_deleted_` so those are not confused?

Reply:
Good point